src/settings.rs

```diff
@@ -13,6 +13,43 @@ pub struct PaginationSetting {
     pub max_total_hits: usize,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct MinWordSizeForTypos {
+    pub one_typo: Option<u8>,
+    pub two_typos: Option<u8>,
+}
+
+impl Default for MinWordSizeForTypos {
+    fn default() -> Self {
+        MinWordSizeForTypos {
+            one_typo: Some(5),
+            two_typos: Some(9),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+#[serde(default)]
+pub struct TypoToleranceSettings {
+    pub enabled: Option<bool>,
+    pub disable_on_attributes: Option<Vec<String>>,
+    pub disable_on_words: Option<Vec<String>>,
+    pub min_word_size_for_typos: Option<MinWordSizeForTypos>,
+}
+
+impl Default for TypoToleranceSettings {
+    fn default() -> Self {
+        TypoToleranceSettings {
+            enabled: Some(true),
+            disable_on_attributes: Some(vec![]),
+            disable_on_words: Some(vec![]),
+            min_word_size_for_typos: Some(MinWordSizeForTypos::default()),
+        }
+    }
+}
```

Comment:
Once again, no reason to force some settings when creating the default value. 

```suggestion
impl Default for TypoToleranceSettings {
    fn default() -> Self {
        TypoToleranceSettings {
            enabled: None,
            disable_on_attributes: None,
            disable_on_words: None,
            min_word_size_for_typos: None,
        }
    }
}
```